question-generation/comments-wrangling/src/taskpane/taskpane.js

```diff
@@ -4,22 +4,36 @@ Office.onReady((info) => {
   if (info.host === Office.HostType.Word) {
     document.getElementById("sideload-msg").style.display = "none";
     document.getElementById("app-body").style.display = "flex";
-    document.getElementById("run").onclick = () => tryCatch(getComments);
+    document.getElementById("run").onclick = () => tryCatch(main);
   }
 });
 
-async function getComments() {
-  await Word.run(async (context) => {
-    // Get comments from the Word document
-    const comments = context.document.body.getComments();
-    comments.load("items");
-    await context.sync();
+async function getCommentContexts(commentCollection) {
+  const commentArray = commentCollection.items;
+  const commentRanges = commentArray.map((comment) => comment.getRange().load("text"));
+  await commentCollection.context.sync();
+  return commentRanges.map((range) => range.text);
+}
 
-    // Store them as an array
-    const commentItems = comments.items;
+async function getCommentReplies(commentCollection) {
+  const commentArray = commentCollection.items;
+  const replyCollections = commentArray.map((comment) => comment.replies.load("items"));
```

Comment:
A lot of repeated code. Leave it for now, but if we find ourselves doing a similar thing a third time, refactor.